packages/client/src/perma-cache.js

```diff
@@ -0,0 +1,396 @@
+        method: 'GET',
+        headers,
+      })
+      const result = await request.json()
+      if (!request.ok) {
+        throw new Error(result.message)
+      }
+
+      // Go over next links until not provided by API anymore
+      const link = request.headers.get('link')
+      if (link) {
+        nextPageUrl = new URL(
+          link.replace('<', '').replace('>; rel="next"', ''),
+          endpoint
+        )
+      } else {
+        // @ts-ignore typescript complains about URL type cannot be undefined
+        nextPageUrl = undefined
+      }
+
+      for (const entry of result) {
+        yield entry
+      }
+    } while (nextPageUrl)
+  }
+
+  /**
+   * @param {Service} service
+   * @param {string[]} urls
+   * @param {DeleteOptions} [options]
+   * @returns {Promise<CacheDeleteResult[]>}
+   */
+  static async delete(
+    { endpoint, token, rateLimiter = globalRateLimiter },
+    urls,
+    { onDelete, maxRetries = MAX_RETRIES } = {}
+  ) {
+    areNftStorageLinkUrls(urls)
+
+    const headers = PermaCache.headers(token)
+
+    /** @type {import('p-settle').PromiseResult<CacheDeleteResult>[]} */
+    const cacheDeleteResults = await pSettle(
+      urls.map(async (url) => {
+        const apiUrl = new URL(
+          `perma-cache/${encodeURIComponent(url)}`,
+          endpoint
+        )
+
+        try {
+          return await pRetry(
+            async () => {
+              await rateLimiter()
+              const request = await fetch(apiUrl.toString(), {
+                method: 'DELETE',
+                headers,
+              })
+
+              const res = await request.json()
+              if (!request.ok) {
+                const e = new Error(res.message)
+                // do not retry if abortable errors - will not succeed
+                if (ABORTABLE_ERRORS_STATUS.includes(request.status)) {
+                  throw new AbortError(e)
+                }
+                /* c8 ignore next 2 */
+                throw e
+              }
+
+              onDelete && onDelete(url)
+
+              return {
+                url,
+              }
+            },
+            /* c8 ignore next 3 */
+            {
+              retries: maxRetries == null ? MAX_RETRIES : maxRetries,
+            }
+          )
+        } catch (error) {
+          return {
+            // @ts-ignore non error type
+            error: error.message,
+            url,
+          }
+        }
+      })
+    )
+
+    // @ts-ignore p-settle will always handle error internally
+    return cacheDeleteResults.map((cacheResult) => cacheResult.value)
+  }
+
+  /**
+   * @param {Service} service
+   * @return {Promise<AccountInfo>}
+   */
+  static async accountInfo({
+    endpoint,
+    token,
+    rateLimiter = globalRateLimiter,
+  }) {
+    const url = new URL('perma-cache/account', endpoint)
+    const headers = PermaCache.headers(token)
+
+    await rateLimiter()
+    const request = await fetch(url.toString(), {
```

Comment:
Retry this also?

Reply:
Well, this should not really fail. The critical one is put because of downloading content... We can add it here too, but I think we should go without it and rethink if it is a problem. Thoughts?